### DISCUSSION TITLE
Feature: Objects/Arrays (Beta)

### DISCUSSION THREAD

**Discussion by kazad** [kazad/instacalc#93] [Announcements]
Having native support for objects/arrays has been on my wishlist for a while. Here's the beta preview:

https://instacalc.com/57570
<img width="927" alt="Screenshot 2024-06-24 at 8 46 11 PM" src="https://github.com/kazad/instacalc/assets/115572/4a283937-9fe3-4e2a-870c-15d9bc186fc1">

Features:
* Define JS-style objects and arrays, using instacalc calculations.
* Access elements using typical format: `obj.field` `array[index]` (starting at `0`)
* Expand/collapse objects for a quick preview of their contents

![image](https://github.com/kazad/instacalc/assets/115572/57a98989-cc76-4440-a783-fa3dff30302d)
![image](https://github.com/kazad/instacalc/assets/115572/78bc879f-94e6-4a16-956f-a5a05c4b9264)

* Use row range functions (`max, min, avg,` etc.) with arrays

This feature is brand new and the behavior may change. Some little gotchas:

* Instacalc is forgiving, so `x. = 3` defines a variable called `x.` which has value `3`. Some people use `.` in variable names.
* Max array size of 1000.
* This paves the way to loading remote data files and manipulating the data in instacalc.

**Comment by EC-SG** [comment]
Arrays - would it be useful to add an increment operator?

items = {val:1++, count:10} ---------> [1, 2 , 3 , 4 , 5 , 6 , 7 , 8 , 9, 10]    // _simplest form & most useful_
or items = {val:1+1, count:10} 

items = {val:1+2, count:10} ---------> [1, 3 , 5 , 7 , 9 , 11 , 13 , 15 , 17, 19]

items = {val:5+5, count:10} ---------> [5, 10 , 15 ,20 , 25 , 30 , 35 , 40 , 45, 50]

this would solve most needs 80/20
without having to get into another (map) function or loops.

And possibly?.... items = {val:2*2, count:5} ---------> [2, 4 , 8 , 16 , 32 ]

**Reply by kazad** [reply]
Good idea; I'll have to think of simple syntax for it, I'd like to see what other languages do.

**Reply by kazad** [reply]
I've built a SEQUENCE() feature, details here: https://github.com/kazad/instacalc/discussions/127
